editor/editor_help_search.cpp

```diff
@@ -394,6 +419,226 @@ bool EditorHelpSearch::Runner::_is_class_disabled_by_feature_profile(const Strin
 	return false;
 }
 
+bool EditorHelpSearch::Runner::_fill() {
+	bool phase_done = false;
+	switch (phase) {
+		case PHASE_MATCH_CLASSES_INIT:
+			phase_done = _phase_fill_classes_init();
+			break;
+		case PHASE_MATCH_CLASSES:
+			phase_done = _phase_fill_classes();
+			break;
+		case PHASE_CLASS_ITEMS_INIT:
+			phase_done = _phase_fill_member_items_init();
+			break;
+		case PHASE_CLASS_ITEMS:
+			phase_done = _phase_fill_member_items();
+			break;
+		case PHASE_MEMBER_ITEMS_INIT:
+		case PHASE_MEMBER_ITEMS:
+		case PHASE_SELECT_MATCH:
+		case PHASE_MAX:
+			return true;
+		default:
+			WARN_PRINT("Invalid or unhandled phase in EditorHelpSearch::Runner, aborting search.");
+			return true;
+	}
+
+	if (phase_done) {
+		phase++;
+	}
+	return false;
+}
+
+bool EditorHelpSearch::Runner::_phase_fill_classes_init() {
+	// Initialize fill.
+	iterator_stack.clear();
+	matched_classes.clear();
+
+	// Initialize stack of iterators to fill, in reverse.
+	iterator_stack.push_back(EditorHelp::get_doc_data()->inheriting[""].back());
+
+	return true;
+}
+
+bool EditorHelpSearch::Runner::_phase_fill_classes() {
+	if (iterator_stack.is_empty()) {
+		return true;
+	}
+
+	if (iterator_stack[iterator_stack.size() - 1]) {
+		DocData::ClassDoc *class_doc = EditorHelp::get_doc_data()->class_list.getptr(iterator_stack[iterator_stack.size() - 1]->get());
+
+		iterator_stack[iterator_stack.size() - 1] = iterator_stack[iterator_stack.size() - 1]->prev();
+		if (!iterator_stack[iterator_stack.size() - 1]) {
+			iterator_stack.resize(iterator_stack.size() - 1);
+		}
+
+		if (!class_doc || class_doc->name.is_empty()) {
+			return false;
+		}
+
+		// If class matches the flags, add it to the matched stack.
+		const bool class_matched =
+				(search_flags & SEARCH_CLASSES) ||
+				((search_flags & SEARCH_CONSTRUCTORS) && !class_doc->constructors.is_empty()) ||
+				((search_flags & SEARCH_METHODS) && !class_doc->methods.is_empty()) ||
+				((search_flags & SEARCH_OPERATORS) && !class_doc->operators.is_empty()) ||
+				((search_flags & SEARCH_SIGNALS) && !class_doc->signals.is_empty()) ||
+				((search_flags & SEARCH_CONSTANTS) && !class_doc->constants.is_empty()) ||
+				((search_flags & SEARCH_PROPERTIES) && !class_doc->properties.is_empty()) ||
+				((search_flags & SEARCH_THEME_ITEMS) && !class_doc->theme_properties.is_empty()) ||
+				((search_flags & SEARCH_ANNOTATIONS) && !class_doc->annotations.is_empty());
+
+		if (class_matched) {
+			matched_classes.push_back(class_doc);
+		}
+
+		// Add inheriting classes, in reverse.
+		if (class_doc && EditorHelp::get_doc_data()->inheriting.has(class_doc->name)) {
+			iterator_stack.push_back(EditorHelp::get_doc_data()->inheriting[class_doc->name].back());
+		}
+
+		return false;
+	}
+
+	if (!iterator_stack[iterator_stack.size() - 1]) {
```

Comment:
Weird? I am not sure what this is for at a glance, nor the order of operations here.

Reply:
It checks if the last entry in the stack has data, it's a standard operation